plugins/utils/nvim-ufo.nix

```diff
@@ -0,0 +1,80 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.nvim-ufo;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.nvim-ufo =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "nvim-ufo";
+
+      package = helpers.mkPackageOption "nvim-ufo" pkgs.vimPlugins.nvim-ufo;
+
+      openFoldHighlightTimeout = helpers.defaultNullOpts.mkInt 400 ''
+        Time in millisecond between the range to be highlgihted and to be cleared
+        while opening the folded line, `0` value will disable the highlight
+      '';
+
+      providerSelector = helpers.defaultNullOpts.mkStr "null" ''
+        A lua function as a selector for fold providers.
+      '';
+
+      closeFoldKinds = helpers.mkNullOrOption types.attrs ''
+        After the buffer is displayed (opened for the first time), close the
+        folds whose range with `kind` field is included in this option. For now,
+        'lsp' provider's standardized kinds are 'comment', 'imports' and 'region',
+        run `UfoInspect` for details if your provider has extended the kinds.
+      '';
+
+      foldVirtualTextHandler = helpers.defaultNullOpts.mkStr "null" "A lua function to customize fold virtual text";
+
+      enableGetFoldVirtualText = helpers.defaultNullOpts.mkBool false ''
+        Enable a function with `lnum` as a parameter to capture the virtual text
+        for the folded lines and export the function to `get_fold_virt_text` field of
+        ctx table as 6th parameter in `fold_virt_text_handler`
+      '';
+
+      preview = {
+        winConfig = {
+          border = helpers.defaultNullOpts.mkStr "rounded" "The border for preview window, `:h nvim_open_win() | call search('border:')`";
```

Comment:
Border can be either a string or a list of 8 strings (maybe not checking the `8` part, but it's usefull to have the list part)